tools/summarize-perf.ps1

```diff
@@ -0,0 +1,53 @@
+param (
+    [Parameter(Mandatory=$true)]
+    [string[]]$Files
+)
+
+Set-StrictMode -Version 'Latest'
+$ErrorActionPreference = 'Stop'
+
+# Important paths.
+$RootDir = Split-Path $PSScriptRoot -Parent
+. $RootDir\tools\common.ps1
+
+$Summary = @{}
+
+# First, merge metrics from all runs, indexed by scenario name, platform, and commit hash.
+foreach ($File in Get-Item -Path $Files) {
+    $Data = New-PerfDataSet -Files $File
+
+    foreach ($Run in $Data) {
+        $Key = [ordered]@{
+            ScenarioName = $Run.ScenarioName
+            Platform = $Run.Platform
+            CommitHash = $Run.CommitHash
+        }
+        $KeyString = $Key | Out-String
+
+        if (!$Summary.ContainsKey($KeyString)) {
+            $Summary[$KeyString] = @{
+                Key = $Key
+                Metrics = @{}
+            }
+        }
+
+        foreach ($Metric in $Run.Metrics) {
+            [array]$Summary[$KeyString].Metrics[$Metric.Name] += $Metric.Value
+        }
+    }
+}
+
+# Second, flatten the merged metrics and summarize the results across runs.
+$Summary.Values | ForEach-Object {
+    foreach ($Metric in $_.Metrics.GetEnumerator()) {
+        [pscustomobject]@{
+            ScenarioName = $_.Key.ScenarioName
+            Platform = $_.Key.Platform
+            CommitHash = $_.Key.CommitHash
```

Comment:
Can we use the short hash instead of the full one?

Reply:
I agree that would be more readable formatting